packages/cart/src/services/shipping-method.ts

```diff
@@ -23,114 +9,10 @@ type InjectedDependencies = {
-  ): Promise<TEntity> {
-    return (await retrieveEntity<ShippingMethod, CartShippingMethodDTO>({
-      id: id,
-      entityName: ShippingMethod.name,
-      repository: this.shippingMethodRepository_,
-      config,
-      sharedContext,
-    })) as TEntity
-  }
-
-  @InjectManager("shippingMethodRepository_")
-  async list(
-    filters: any = {},
-    config: FindConfig<CartShippingMethodDTO> = {},
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<TEntity[]> {
-    const queryOptions = ModulesSdkUtils.buildQuery<ShippingMethod>(
-      filters,
-      config
-    )
-
-    return (await this.shippingMethodRepository_.find(
-      queryOptions,
-      sharedContext
-    )) as TEntity[]
-  }
-
-  @InjectManager("shippingMethodRepository_")
-  async listAndCount(
-    filters: any = {},
-    config: FindConfig<CartShippingMethodDTO> = {},
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<[TEntity[], number]> {
-    const queryOptions = ModulesSdkUtils.buildQuery<ShippingMethod>(
-      filters,
-      config
-    )
-
-    return (await this.shippingMethodRepository_.findAndCount(
-      queryOptions,
-      sharedContext
-    )) as [TEntity[], number]
-  }
-
-  @InjectTransactionManager("shippingMethodRepository_")
-  async create(
-    data: CreateShippingMethodDTO[],
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<TEntity[]> {
-    return (await (
-      this.shippingMethodRepository_ as ShippingMethodRepository
-    ).create(data, sharedContext)) as TEntity[]
+> extends ModulesSdkUtils.abstractServiceFactory<
+  InjectedDependencies,
+  {
+    create: CreateShippingMethodDTO
+    update: UpdateShippingMethodDTO
   }
-
-  @InjectTransactionManager("shippingMethodRepository_")
-  async update(
-    data: UpdateShippingMethodDTO[],
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<TEntity[]> {
-    const existingMethods = await this.list(
-      {
-        id: data.map((d) => d.id),
-      },
-      {},
-      sharedContext
-    )
-
-    const existingMethodsMap = new Map(
-      existingMethods.map<[string, ShippingMethod]>((sm) => [sm.id, sm])
-    )
-
-    const updates: UpdateShippingMethodDTO[] = []
-
-    for (const update of data) {
-      const shippingMethod = existingMethodsMap.get(update.id)
-
-      if (!shippingMethod) {
-        throw new MedusaError(
-          MedusaError.Types.NOT_FOUND,
-          `Shipping method with id "${update.id}" not found`
-        )
-      }
-
-      updates.push({ ...update, id: shippingMethod.id })
-    }
-
-    return (await (
-      this.shippingMethodRepository_ as ShippingMethodRepository
-    ).update(updates, sharedContext)) as TEntity[]
-  }
-
-  @InjectTransactionManager("shippingMethodRepository_")
-  async delete(
-    ids: string[],
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<void> {
-    await this.shippingMethodRepository_.delete(ids, sharedContext)
-  }
-}
+>(ShippingMethod)<TEntity> {}
```

Comment:
I suggest to put the constructor. The moment you want to add unit tests, if there is no constructor you will have some issues since typescript will tell you that it doesnt expect any arguments even though it expects the container. wdyt?